### DISCUSSION TITLE
Issue templates, Labels, Projects & Milestones

### DISCUSSION THREAD

**Discussion by hasanmanzak** [Biarity/Sieve#130] [General]
I think we should prepare some templates to provide people to be able to communicate more smoothly.

Currently open issues are about
* either something is working but the use case is different and the author asks about that feature to work the other way,
* or something is not working and 
  * either it is actually not a feature of the library and if we put a work on then it would be a nice addition to the library,
  * or is a bug.

All of those are very hard to follow and seperate and manage and etc etc.

For example, I would like to discuss these issues with the maintainers:
#2 
#10 
#37 
#77 
#104 
#106 
#120 

Last but not least, not only we need the templates but we can also benefit from the projects board and milestones aswell. Issue lifetime management, PR management, Preview releases, targeted releases.. C'mon! :)

**Comment by ITDancer13** [comment]
I think that's a great idea. I'm setting up a pipeline based on GitHub Actions to be able to publish (pre-)releases automatically. As soon as this works, I'd like to screen all open issues. As I'm not familiar with the features of managing issues and projects in GitHub it would be great if you (or someone else) can assist me with that.

**Comment by ITDancer13** [comment]
@hasanmanzak according to the other issue, @Biarity invited you to maintain Sieve, too. Is that correct? If yes, I think we're a couple of maintainers now and we should sync on what to do next and what do we want to solve for the next release.

**Comment by Biarity** [comment]
I've added a basic issue template for bugs. For new feature requests and other discussion we'll use [the discussions tab](https://github.com/Biarity/Sieve/discussions). We can use upvotes there as a means of prioritization. 

We should close any issues that are feature requests and re-open them as discussions. My next priority is to close off as much stale PRs/issues as possible and ensure some stability so we can safely focus on new features. I want to have a v2.5 where we do that, then perhaps brainstorm some new ideas for v3. 

I see this project headed towards becoming a lightweight, more customizeable OData/GraphQL alternative, so next steps may include writing a more efficient parser, writing a JS client library, or adding select/search/etc. Any thoughts?

**Reply by ITDancer13** [reply]
That sounds great. I move my board in the project section from Pagr to Sieve - everything is open for discussion ;-)
But I think it would be good to cleanup the existing issues by solving some quick fixes in a next version. I'll start with some integration test projects based on .NET5 and .NET6 in the next days to ensure that we're ready for the next LTS of .NET.

**Reply by Biarity** [reply]
No that looks good! I'm happy with throwing exceptions by default, seems to be the only breaking one, so would make sense to have it be part of v3

**Comment by Biarity** [comment]
I've added v2.5 and v3.0 milestones - feel free to assign issues there to define scope

**Reply by ITDancer13** [reply]
Shall I create a `releases/2.5.0` branch with all required changes for the pipeline to release previews and the final version automatically or do you want to do it manually this time?

**Reply by hasanmanzak** [reply]
I think a 2.5.0 branch would do nice. We just do a cleanup a bit and see where we're heading next. And also this would be the first fully featured, automated release test for us before going for version 3 and would direct us what to manage, how to manage. What do you think?

**Reply by ITDancer13** [reply]
Branch is created and setup. We can now merge all changes for version 2.5 into this branch.

**Reply by hasanmanzak** [reply]
MAN! You are fast :)) Should I tag current PRs to target the version 2.5 then?

**Reply by Biarity** [reply]
Yeah I think current PRs are a good start for 2.5
